File: mynote-android/app/src/main/java/com/xuyazhou/mynote/model/db/AttachMent.java
package com.xuyazhou.mynote.model.db;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

/**
 * Author: lampard_xu(xuyazhou18@gmail.com)
 * *
 * Date: 2016-12-19
 */

@Table(database = AppDatabase.class)
public class AttachMent extends BaseModel {



    @Column
    String noteId;

    @PrimaryKey
    @Column
    String sid;


    @Column
    String descriprion;

    @Column
    String localPath;

    @Column
    String fileType;

    @Column
    String fileName;

    @Column
    long size;

    @Column
    long createTime;

    @Column
    long modifiedTime;

    @Column
    int status;

    @Column
    boolean deleted;

    @Column
    String spath;





    public String getSid() {
        return sid;
    }

    public void setSid(String sid) {
        this.sid = sid;
    }

    public long getCreateTime() {
        return createTime;
    }

    public void setCreateTime(long createTime) {
        this.createTime = createTime;
    }

    public String getNoteId() {
        return noteId;
    }

    public void setNoteId(String noteId) {
        this.noteId = noteId;
    }


    public String getDescriprion() {
        return descriprion;
    }

    public void setDescriprion(String descriprion) {
        this.descriprion = descriprion;
    }

    public String getLocalPath() {
        return localPath;
    }

    public void setLocalPath(String localPath) {
        this.localPath = localPath;
    }

    public String getFileType() {
        return fileType;
    }

    public void setFileType(String fileType) {
        this.fileType = fileType;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public long getSize() {
        return size;
    }

    public void setSize(long size) {
        this.size = size;
    }

    public long getCrateTime() {
        return createTime;
    }

    public void setCrateTime(long crateTime) {
        this.createTime = crateTime;
    }

    public long getModifiedTime() {
        return modifiedTime;
    }

    public void setModifiedTime(long modifiedTime) {
        this.modifiedTime = modifiedTime;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public boolean isDeleted() {
        return deleted;
    }

    public void setDeleted(boolean deleted) {
        this.deleted = deleted;
    }


    public String getSpath() {
        return spath;
    }

    public void setSpath(String spath) {
        this.spath = spath;
    }




}

File: mynote-ios/MyNote-Ios/Models/db/AttachMent.swift
//
//  AttachMent.swift
//  MyNote-Ios
//
//  Created by xuyazhou on 2017/9/8.
//  Copyright © 2017年 xuyazhou. All rights reserved.
//

import RealmSwift
import ObjectMapper

class AttachMent: Object, Mappable {


    dynamic var noteId = ""
    dynamic var fileName = ""
    dynamic var fileData: Data?
    dynamic var fileType = ""
    dynamic var spath = ""
    dynamic var sid = ""
    dynamic var createTime = 0
    dynamic var modifiedTime = 0
    dynamic var status = 0
    dynamic var size = 0
    dynamic var deleted = false

    override static func primaryKey() -> String? {
        return "sid"
    }

    required convenience init?(map: Map) {
        self.init()
    }

    func mapping(map: Map) {
        noteId <- map["noteId"]
        fileName <- map["fileName"]
        fileType <- map["fileType"]
        spath <- map["spath"]
        createTime <- (map["createTime"])
        modifiedTime <- (map["modifiedTime"])

        if map.mappingType == .toJSON {
            var sid = self.sid
            sid <- map["sid"]
        } else {
            sid <- map["sid"]
        }

        size <- map["size"]
        deleted <- map["deleted"]
        status <- map["status"]

    }
}
